### PR TITLE
gh-91832: Add 'required' attr to argparse.Action repr

### PR DESCRIPTION
**Pull request by AbhigyanBose**
# Adding 'required' to names in Lib.argparse.Action

gh-91832: 
Added 'required' to the list `names` in `Lib.argparse.Action`. 
Changed constant strings that test the Action object.

Automerge-Triggered-By: GH:merwok

### CODE REVIEW

**Comment by fatihkabakk** [comment]
We got the same error, here is my pr #91840.
It's weird that when patchcheck made it deletes whitespaces, but when not it throws error.
I also mentioned that problem on my comment in #91840 .

**Comment by AbhigyanBose** [comment]
Thanks for pointing it out. I just realized the same.

File: Lib/test/test_argparse.py [hunk 1]
```diff
@@ -2265,7 +2265,7 @@ def test_help_blank(self):
             main description
 
             positional arguments:
-              foo         
+              foo
```
Comment (AbhigyanBose) [inline Lib/test/test_argparse.py]: Done by `make patchcheck`

File: Lib/test/test_argparse.py [hunk 2]
```diff
@@ -2282,7 +2282,7 @@ def test_help_blank(self):
             main description
 
             positional arguments:
-              {}          
+              {}
```
Comment (AbhigyanBose) [inline Lib/test/test_argparse.py]: Done by `make patchcheck`

File: Lib/test/test_argparse.py [hunk 3]
```diff
@@ -2265,7 +2265,7 @@ def test_help_blank(self):
             main description
 
             positional arguments:
-              foo
+              foo
```
Comment (AbhigyanBose) [inline Lib/test/test_argparse.py]: Reverting changes made by `make patchcheck`

File: Lib/test/test_argparse.py [hunk 4]
```diff
@@ -2282,7 +2282,7 @@ def test_help_blank(self):
             main description
 
             positional arguments:
-              {}
+              {}
```
Comment (AbhigyanBose) [inline Lib/test/test_argparse.py]: Reverting changes made by `make patchcheck`

File: Misc/NEWS.d/next/Library/2022-04-23-03-24-00.gh-issue-91832.TyLi65.rst [hunk 5]
```diff
@@ -0,0 +1 @@
+Added 'required' to names list in Lib.argparse.Action.
```
Comment (merwok) [inline Misc/NEWS.d/next/Library/2022-04-23-03-24-00.gh-issue-91832.TyLi65.rst]: Python stdlib doesn’t use a root package name, Lib is just a source directory.

```suggestion
Added 'required' to names list in argparse.Action.
```

Also mentioning `parameters` may be clearer than `names`

Comment (AbhigyanBose) [reply Misc/NEWS.d/next/Library/2022-04-23-03-24-00.gh-issue-91832.TyLi65.rst]: `names` is the name of the list in Action class that we are adding the element to. I should've put it in a code block. 

How does this look ?
"Added an element `required` to the list `names` in argparse.Action._get_kwargs"

Comment (merwok) [reply Misc/NEWS.d/next/Library/2022-04-23-03-24-00.gh-issue-91832.TyLi65.rst]: Same thing.  My point is that it does not matter that a line in the code uses `names`, it is not meaninfgul on its own when read by people in the NEWS.

Comment (AbhigyanBose) [reply Misc/NEWS.d/next/Library/2022-04-23-03-24-00.gh-issue-91832.TyLi65.rst]: How does this sound then:

Added a parameter "required" to the list returned by `_get_kwargs` in `argparse.Action`.

Comment (merwok) [reply Misc/NEWS.d/next/Library/2022-04-23-03-24-00.gh-issue-91832.TyLi65.rst]: The bug really is: required is missing from repr output
So this fix is doing: `Add 'required' attr to argparse.Action repr`
🙂

Comment (AbhigyanBose) [reply Misc/NEWS.d/next/Library/2022-04-23-03-24-00.gh-issue-91832.TyLi65.rst]: Ahhh..... Sorry, I'm still new here, and trying to get to know things. Thanks for taking the time to help me understand.

Comment (merwok) [reply Misc/NEWS.d/next/Library/2022-04-23-03-24-00.gh-issue-91832.TyLi65.rst]: No worry!  There is info in the devguide: https://devguide.python.org/pullrequest/

(for future PRs, please don’t force push, it makes reviews harder)

**Comment by terryjreedy** [comment]
As suggested by Ronald Oussoren on core mentorship, changing a line like '   x   ' to '   x   \n\' should solve the patchcheck failure.  

In a future issue, parser.format_help might be modified to not format with trailing whitespace.  I doubt that it is really needed.

Sidenote: I am not sure what you did, but force-pushing is rarely needed when revising PRs and sometimes messes them up.

**Comment by AbhigyanBose** [comment]
@terryjreedy Got it. I just wasn't sure if I should change an unrelated test case's content in this particular PR. If that's what you and @merwok recommend I'll add in '\n' and then run `make patchcheck` before I submit. 

I could also create a separate PR to add '\n' to that test case.

**Comment by terryjreedy** [comment]
A separate PR would be OK.   Azure Pipelines passing is not required but its failure from patchcheck failing is a nuisance.

**Comment by terryjreedy** [comment]
I think Zachary Ware's suggestion of '\x20' might be even better.  (How to fix argparse is not obvious.  It uses %-formatting and I check all %s and no of format string I looked at had trailing spaces of format character that would produce such.)

**Comment by merwok** [comment]
This format spec does funky alignment: https://github.com/python/cpython/blob/94d0853b0437755a11d47cddee1f32ae1c3cc5dd/Lib/argparse.py#L521

**Comment by AbhigyanBose** [comment]
@merwok  and @terryjreedy  

I created a new issue for this. I also proposed the solution I thought to be the simplest there.

**Comment by AbhigyanBose** [comment]
https://github.com/python/cpython/issues/91984 has been resolved, thus `make patchcheck` is successful. Is there any other change required in this PR ?

File: Misc/NEWS.d/next/Library/2022-04-23-03-24-00.gh-issue-91832.TyLi65.rst [hunk 6]
```diff
@@ -0,0 +1 @@
+Add 'required' attr to argparse.Action repr.
```
Comment (merwok) [inline Misc/NEWS.d/next/Library/2022-04-23-03-24-00.gh-issue-91832.TyLi65.rst]: News entries don’t have to be as terse as PR titles (which become commit messages, so should follow common conventions for commit messages), and use reST markup.

```suggestion
Add ``required`` attribute to :class:`argparse.Action` repr output.
```

**Review by merwok — COMMENTED** [review]
Thanks for the PR!

Minor change in news entry then this is good to go.
Can you check if backports to 3.10 and 3.9 are needed?  (if the `required` attribute is present in these versions too)

**Review by merwok — APPROVED** [review]

**Pull request merged**